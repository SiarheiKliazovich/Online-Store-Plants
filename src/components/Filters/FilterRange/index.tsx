import { FunctionComponent } from "react";
import { FilterRangeType } from "../../../types";

const FilterRange: FunctionComponent<FilterRangeType> = ({
  minLabel,
  maxLabel,
  valueFromSlider,
  minFromSlider,
  maxFromSlider,
  setFiltersFromSlider,
  setFiltersToSlider,
  valueToSlider,
  minToSlider,
  maxToSlider,
}: FilterRangeType) => (
  <div className="filter__list">
    <div className="form_control">
      <div className="form_control_container__time">{minLabel}</div>‚ü∑
      <div className="form_control_container__time">{maxLabel}</div>
    </div>
    <div className="range_container">
      <div className="sliders_control">
        <input
          id="fromSlider"
          type="range"
          value={valueFromSlider}
          min={minFromSlider}
          max={maxFromSlider}
          onChange={(event) => setFiltersFromSlider(event.target.value)}
        />
        <input
          id="toSlider"
          type="range"
          value={valueToSlider}
          min={minToSlider}
          max={maxToSlider}
          onChange={(event) => setFiltersToSlider(event.target.value)}
        />
      </div>
    </div>
  </div>
);

export default FilterRange;
