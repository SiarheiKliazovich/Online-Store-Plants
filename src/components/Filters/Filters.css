.filter {
  padding-top: 15px;
}

.filter__panel {
  background-color: #000;
  border-radius: 4px;
  margin-bottom: 10px;
}

.filter__panel_text {
  color: #ffffff;
  padding: 10px 15px;
  text-transform: uppercase;
}

.filter__color {
  display: flex;
  flex-direction: column;
}

.filter__color_item {
  border-bottom: 1px solid #e4e4e4;
  padding: 10px 0 10px 10px;
  display: flex;
  justify-content: space-between;
}

.color-item__link {
  text-decoration: none;
  cursor: pointer;
  color: #000;
}

/*# sourceMappingURL=Filters.css.map */
