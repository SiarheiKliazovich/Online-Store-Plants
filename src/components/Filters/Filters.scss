@import "../../sass/variables";

.filter {
  padding-top: 15px;
}

.filter__panel {
  background-color:$color-black;
  border-radius: 4px;
  margin-bottom: 10px;
  display: flex;
  justify-content: space-around;
}

.btn__filters-reset {
  padding: 5px 5px;
  margin: 10px 2px;
  font-family: $font-family-primary;
  background-color:$color-green-black;
  color: $color-white;
  border-color:$color-grey;
  border-radius: 5px;
  cursor: pointer;

  &:hover {
    background-color: $color-green-hover;
  }
}

.btn__copy-link {
  padding: 5px 5px;
  margin: 10px 2px;
  font-family: $font-family-primary;
  background-color:$color-green-black;
  color: $color-white;
  cursor: pointer;
  color: $color-white;
  border-color:$color-grey;
  border-radius: 5px;
  
  &:hover {
    background-color: $color-green-hover;
  }
}

// .filter__panel_text {
//   color: $color-white;
//   padding: 10px 15px;
//   text-transform: uppercase;
// }

// .filter__color {
//   display: flex;
//   flex-direction: column;
// }

// .filter__color_item{
//   border-bottom: 1px solid $color-grey;
//   padding:10px 0 10px 10px;
//   display: flex;
//   justify-content: space-between;
// }

// .color-item__link {
//   text-decoration: none;
//   cursor: pointer;
//   color: $color-black;
// }

.category__wrapper {
border: 2px solid $color-grey;
border-radius: 5px;
overflow: hidden;
margin-bottom: 5%;
display: block;
}

.title__filter {
  text-align: center;
  background-color:$color-black;
  border-bottom: 2px solid $color-grey;
  color: $color-white;
  font-family: $font-family-primary;
  padding: 3px;
}

.filter__list {
  overflow: hidden;
  overflow-y: hidden;
  overflow-y: auto;
 
}

.checkbox-filter {
  white-space: nowrap;
  display: flex;
  align-items: center;
}

.input__filter {
  cursor: grab;
  box-sizing: border-box;
  font-size: calc(.2rem + 1vw);
  color: orange;
  font-family: $font-family-primary;
  word-wrap: break-word;
}

.label__filter {
  overflow: hidden;
  padding: 5px;
  cursor: pointer;
  -webkit-user-select: none;
  user-select: none;
}

.span__filter {
  margin-left: auto;
  padding-left: .3rem;
}

.brand__wrapper {
  border: 2px solid $color-grey;
  border-radius: 5px;
  overflow: hidden;
  margin-bottom: 5%;
  display: block;
}

.price__wrapper {
  border: 2px solid $color-grey;
  border-radius: 5px;
  overflow: hidden;
  margin-bottom: 10px;
  display: block;
}

.stock__wrapper {
  border: 2px solid $color-grey;
  border-radius: 5px;
  overflow: hidden;
  margin-bottom: 10px;
  display: block;
}