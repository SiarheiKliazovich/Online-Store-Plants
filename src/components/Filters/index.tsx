import "./Filters.scss";
import products from "../../data/products";

const filterItem = () => {
  const newBrand: string[] = [];
  const newPrice: number[] = [];
  const newCateg: string[] = [];
  const newStock: number[] = [];
  products.map((product) => newCateg.push(product.category));
  products.map((product) => newBrand.push(product.brand));
  products.map((product) => newPrice.push(product.price));
  products.map((product) => newStock.push(product.stock));
  const Category = newCateg.filter((element, index) => {
    return newCateg.indexOf(element) === index;
  });
  const Brand = newBrand.filter((element, index) => {
    return newBrand.indexOf(element) === index;
  });
  const Price = newPrice.filter((element, index) => {
    return newPrice.indexOf(element) === index;
  });
  const Stock = newStock.filter((element, index) => {
    return newStock.indexOf(element) === index;
  });
  /*
  {category:{list:[],all:0, filtered:0}, brand:[], price:[],stock:[]}
  */
  return {
    Category,
    Brand,
    Price,
    Stock,
  };
};

const Filter = () => {
  return (
    <div className="filter">
      <div className="filter__panel">
        <button className="btn__filters-reset">Reset Filters</button>
        <button className="btn__copy-link">Copy Link</button>
      </div>
      <div className="category__wrapper">
        <h3 className="title__filter">Category</h3>
        {filterItem().Category.map((category) => (
          <div className="filter__list">
            <div className="checkbox-filter">
              <input
                value={category}
                className="input__category"
                type="checkbox"
              />
              <label className="label__category" htmlFor="">
                {category}
              </label>
              <span className="span__category">(?/?)</span>
            </div>
          </div>
        ))}
      </div>
      <div className="brand__wrapper">
        <h3 className="title__filter">Brand</h3>
        {filterItem().Brand.map((brand) => (
          <div className="filter__list">
            <div className="checkbox-filter">
              <input className="input__filter" type="checkbox" />
              <label className="label__filter" htmlFor="London-Plants&Co">
                {brand}
              </label>
              <span className="span__filter">(?/?)</span>
            </div>
          </div>
        ))}
      </div>
      <div className="price__wrapper">
        <h3 className="title__filter">Price</h3>
        <div className="filter__list"></div>
      </div>
      <div className="stock__wrapper">
        <h3 className="title__filter">Stock</h3>
        <div className="filter__list"></div>
      </div>
    </div>
  );
};
export default Filter;
