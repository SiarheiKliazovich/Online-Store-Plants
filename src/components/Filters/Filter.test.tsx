import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import Filters from "../Filters";

it("Filters component", () => {
  const filters = {
    categories: ["Aeonium"],
    brands: ["London-Plants&Co"],
    prices: [22, 30],
    stocks: [45, 68],
  };
  const setFilters = jest.fn();
  const productList = [
    {
      id: 1,
      name: "Biotech",
      title: "The growth of a plant involves several stages.",
      description: ["Fast, a plot of land is selected."],
      size: "XL",
      color: "Pink",
      availability: "In Stock",
      price: 25,
      category: "Aeonium",
      stock: 3,
      rating: 4,
      brand: "London-Plants&Co",
      thumbnail:
        "https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp",
      images: [
        "https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp",
      ],
    },
  ];
  const filterReset = jest.fn();
  const productFilters = {
    categories: [{ title: "Aeonium", filtered: 2, allProductsCount: 4 }],
    brands: [{ title: "London-Plants&Co", filtered: 3, allBrandCount: 4 }],
    prices: [22, 30],
    stocks: [45, 50],
  };
  const wrapper = render(
    <Router>
      <Filters
        filters={filters}
        setFilters={setFilters}
        productList={productList}
        filterReset={filterReset}
        productFilters={productFilters}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
