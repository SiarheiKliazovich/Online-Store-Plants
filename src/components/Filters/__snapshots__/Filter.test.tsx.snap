// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter"
      >
        <div
          class="filter__btn_wrapper"
        >
          <button
            class="btn__filters-reset"
          >
            Reset Filters
          </button>
          <button
            class="btn__copy-link"
          >
            Copy link
          </button>
        </div>
        <div
          class="filter__panel"
        >
          <h3
            class="title__filter"
          >
            Category
          </h3>
          <div
            class="filter__list"
          >
            <div
              class="checkbox-filter"
            >
              <input
                checked=""
                class="input__category"
                id="Aeonium"
                type="checkbox"
                value="Aeonium"
              />
              <label
                class="label__filter"
                for="Aeonium"
              >
                Aeonium
              </label>
              <span
                class="span__filter"
              >
                (
                1
                /
                4
                )
              </span>
            </div>
          </div>
        </div>
        <div
          class="filter__panel"
        >
          <h3
            class="title__filter"
          >
            Brand
          </h3>
          <div
            class="filter__list"
          >
            <div
              class="checkbox-filter"
            >
              <input
                checked=""
                class="input__filter"
                id="London-Plants&Co"
                type="checkbox"
                value="London-Plants&Co"
              />
              <label
                class="label__filter"
                for="London-Plants&Co"
              >
                London-Plants&Co
              </label>
              <span
                class="span__filter"
              >
                (
                1
                /
                4
                )
              </span>
            </div>
          </div>
        </div>
        <div
          class="filter__panel"
        >
          <h3
            class="title__filter"
          >
            Price
          </h3>
          <div
            class="filter__list"
          >
            <div
              class="form_control"
            >
              <div
                class="form_control_container__time"
              >
                22
              </div>
              ⟷
              <div
                class="form_control_container__time"
              >
                30
              </div>
            </div>
            <div
              class="range_container"
            >
              <div
                class="sliders_control"
              >
                <input
                  id="fromSlider"
                  max="30"
                  min="22"
                  type="range"
                  value="22"
                />
                <input
                  id="toSlider"
                  max="30"
                  min="22"
                  type="range"
                  value="30"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="filter__panel"
        >
          <h3
            class="title__filter"
          >
            Stock
          </h3>
          <div
            class="filter__list"
          >
            <div
              class="form_control"
            >
              <div
                class="form_control_container__time"
              >
                45
              </div>
              ⟷
              <div
                class="form_control_container__time"
              >
                68
              </div>
            </div>
            <div
              class="range_container"
            >
              <div
                class="sliders_control"
              >
                <input
                  id="fromSlider"
                  max="50"
                  min="45"
                  type="range"
                  value="45"
                />
                <input
                  id="toSlider"
                  max="50"
                  min="45"
                  type="range"
                  value="68"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter"
    >
      <div
        class="filter__btn_wrapper"
      >
        <button
          class="btn__filters-reset"
        >
          Reset Filters
        </button>
        <button
          class="btn__copy-link"
        >
          Copy link
        </button>
      </div>
      <div
        class="filter__panel"
      >
        <h3
          class="title__filter"
        >
          Category
        </h3>
        <div
          class="filter__list"
        >
          <div
            class="checkbox-filter"
          >
            <input
              checked=""
              class="input__category"
              id="Aeonium"
              type="checkbox"
              value="Aeonium"
            />
            <label
              class="label__filter"
              for="Aeonium"
            >
              Aeonium
            </label>
            <span
              class="span__filter"
            >
              (
              1
              /
              4
              )
            </span>
          </div>
        </div>
      </div>
      <div
        class="filter__panel"
      >
        <h3
          class="title__filter"
        >
          Brand
        </h3>
        <div
          class="filter__list"
        >
          <div
            class="checkbox-filter"
          >
            <input
              checked=""
              class="input__filter"
              id="London-Plants&Co"
              type="checkbox"
              value="London-Plants&Co"
            />
            <label
              class="label__filter"
              for="London-Plants&Co"
            >
              London-Plants&Co
            </label>
            <span
              class="span__filter"
            >
              (
              1
              /
              4
              )
            </span>
          </div>
        </div>
      </div>
      <div
        class="filter__panel"
      >
        <h3
          class="title__filter"
        >
          Price
        </h3>
        <div
          class="filter__list"
        >
          <div
            class="form_control"
          >
            <div
              class="form_control_container__time"
            >
              22
            </div>
            ⟷
            <div
              class="form_control_container__time"
            >
              30
            </div>
          </div>
          <div
            class="range_container"
          >
            <div
              class="sliders_control"
            >
              <input
                id="fromSlider"
                max="30"
                min="22"
                type="range"
                value="22"
              />
              <input
                id="toSlider"
                max="30"
                min="22"
                type="range"
                value="30"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="filter__panel"
      >
        <h3
          class="title__filter"
        >
          Stock
        </h3>
        <div
          class="filter__list"
        >
          <div
            class="form_control"
          >
            <div
              class="form_control_container__time"
            >
              45
            </div>
            ⟷
            <div
              class="form_control_container__time"
            >
              68
            </div>
          </div>
          <div
            class="range_container"
          >
            <div
              class="sliders_control"
            >
              <input
                id="fromSlider"
                max="50"
                min="45"
                type="range"
                value="45"
              />
              <input
                id="toSlider"
                max="50"
                min="45"
                type="range"
                value="68"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
